src/trackers.js

```diff
@@ -1,227 +1,255 @@
+        } else {
+            return false
         }
-
-        return false
     }
 
-    /* Compare two urls to determine if they came from the same hostname
-     * pull off any subdomains before comparison.
-     * Return parent company name from entityMap if one is found or unknown
-     * if domains match but we don't have this site in our entityMap.
-     */
-    getCommonParentEntity (currLocation, urlToCheck) {
-        if (!entityMap) return
-        let currLocationDomain = utils.getDomain(currLocation)
-        let urlToCheckDomain = utils.getDomain(urlToCheck)
-        let parentEntity = entityMap[urlToCheckDomain]
-        if (currLocationDomain === urlToCheckDomain ||
-                this.isRelatedEntity(parentEntity, currLocation)) {
-            return parentEntity || currLocationDomain
+    /* Check the matched rule exceptions against the request data
+     * Both domains and types need to match (if they exist) in order
+     * for the exception to match.
+    */
+    matchesException (tracker, requestData) {
+        if (tracker.matchedRule && tracker.matchedRule.exceptions) {
+            // all exception types need to match for this to return true
+            return !this.exceptionTypes.some(exceptionType => {
+                const requestParamToCheck = exceptionType === 'domains' ? requestData.siteDomain : requestData.request.type
+                return !(tracker.matchedRule.exceptions[exceptionType] &&
+                          tracker.matchedRule.exceptions[exceptionType].length &&
+                          tracker.matchedRule.exceptions[exceptionType].includes(requestParamToCheck))
+            })
+        } else {
+            return false
         }
-
-        return false
     }
 
-    getTrackerDetails (trackerUrl, listName) {
-        let host = utils.extractHostFromURL(trackerUrl)
-        let parentCompany = utils.findParent(host.split('.')) || 'unknown'
-        return {
-            parentCompany: parentCompany,
-            url: host,
-            type: listName
+    /* Check the matched rule  options against the request data
+    *  return: true (all options matched)
+    */
+    matchRuleOptions (rule, requestData) {
+        if (!rule.options) return true
+
+        if (rule.options.types &&
+            rule.options.types.length &&
+            !rule.options.types.includes(requestData.request.type)) {
+            return false
+        }
+
+        if (rule.options.domains &&
+            rule.options.domains.length &&
+            !rule.options.domains.includes(requestData.siteDomain)) {
+            return false
         }
+
+        return true
     }
 
-    checkABPParsedList (list, url, currLocationDomain, requestType) {
-        let match = abp.matches(list, url,
-            {
-                domain: currLocationDomain,
-                elementTypeMask: abp.elementTypes[requestType.toUpperCase()]
-            })
-        return match
+    setAction (tracker, requestData) {
```

Comment:
this method is pretty awesome 🙌I like how we now have one center piece where everything comes togheter

Reply:
nit: `requestData` is not used here